qp_shogun/shogun/shogun.py

```diff
@@ -198,42 +199,49 @@ def shogun(qclient, job_id, parameters, out_dir):
             return False, None, msg
 
         # Step 4 taxonomic profile
-        sys_msg = "Step 4 of 5: Taxonomic profile with Shogun (%d/{0})"
+        sys_msg = "Step 4 of 6: Taxonomic profile with Shogun (%d/{0})"
         assign_cmd, profile_fp = generate_shogun_assign_taxonomy_commands(
             temp_dir, parameters)
         success, msg = _run_commands(
             qclient, job_id, assign_cmd, sys_msg, 'Shogun taxonomy assignment')
         if not success:
             return False, None, msg
 
-        sys_msg = "Step 5 of 5: Converting output to BIOM"
+        sys_msg = "Step 5 of 6: Converting output to BIOM"
         qclient.update_job_step(job_id, msg)
         output = run_shogun_to_biom(profile_fp, [None, None, None, True],
                                     out_dir, 'profile')
 
         ainfo = [ArtifactInfo('Shogun Alignment Profile', 'BIOM',
                               [(output, 'biom')])]
 
+        # Step 5 redistribute profile
+        sys_msg = "Step 6 of 6: Redistributed profile with Shogun (%d/{0})"
+        levels = ['phylum', 'genus', 'species']
+        redist_fps = []
+        for level in levels:
+            redist_cmd, output = generate_shogun_redist_commands(
+                profile_fp, temp_dir, parameters, level)
+            redist_fps.append(output)
+            success, msg = _run_commands(
+                qclient, job_id, redist_cmd, sys_msg, 'Shogun redistribute')
+            if not success:
+                return False, None, msg
+        # Converting redistributed files to biom
+        for redist_fp, level in zip(redist_fps, levels):
+            biom_in = ["redist", None, '', True]
+            output = run_shogun_to_biom(
+                redist_fp, biom_in, out_dir, level, 'redist')
+            aname = 'Taxonomic Predictions - %s' % level
+            ainfo.append(ArtifactInfo(aname, 'BIOM', [(output, 'biom')]))
+
     return True, ainfo, ""
 
 #
 # This code is not currently needed but it will be used for analysis, so
 # leaving here to avoid having to rewrite it
```

Comment:
Is the expectation that this commented code will be staying in this form temporarily? I'm wary that if I accept it now, it will stay here indefinitely.

Reply:
Staying there forever is a possibility but I don't think it will happen. As you can see some parts of that code were moved to be funcional and other stayed commented as since the beginning we decided that we will limit what was possible. Currently we decided that only taxonomy will be available and functional will be in the future ...